### PR TITLE
[MM-29561] Added e2e tests to the "listCommandCmdF

### PR DESCRIPTION
**Pull request by SezalAgrawal**
#### Summary

Added e2e tests to the "listCommandCmdF" mmctl command

#### Ticket Link

Fixes: mattermost/mattermost-server#15939

JIRA: https://mattermost.atlassian.net/browse/MM-29561

### CODE REVIEW

**Review by streamer45 — CHANGES_REQUESTED** [review]
Thank you for your contribution @SezalAgrawal :+1: 
I'd suggest adding a couple more test cases:

- One to trigger a permission error (listing commands on a team the user doesn't have access to).
- One where we don't pass any team as argument and expect commands for all available teams to be listed.

**Comment by SezalAgrawal** [comment]
Thanks @streamer45 for your suggestions. I will add these test cases.

**Comment by SezalAgrawal** [comment]
@streamer45 I have added the test cases.

**Review by streamer45 — APPROVED** [review]
Thank you @SezalAgrawal . Looks great :+1:

**Review by lindy65 — APPROVED** [review]
Thanks @SezalAgrawal - looks good to me 👍

File: commands/command_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		}()
+
+		err := listCommandCmdF(c, &cobra.Command{}, []string{team.Id})
+		s.Require().Nil(err)
+		s.Len(printer.GetLines(), 1)
+		s.Equal(command, printer.GetLines()[0])
+		s.Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("List all commands from all teams", func() {
```
Comment (mgdelacroix) [inline commands/command_e2e_test.go:65]: This test case should run for both local mode and sysadmin. You can leave this `s.Run` untouched, set up the teams and commands inside and at the end, create an `s.RunForSystemAdminAndLocal` to perform the search with both users with the generated teams

File: commands/command_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	"github.com/spf13/cobra"
+
+	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
+)
+
+func (s *MmctlE2ETestSuite) TestListCommandCmd() {
+	s.SetupTestHelper().InitBasic()
+
+	s.RunForSystemAdminAndLocal("List commands for a non existing team", func(c client.Client) {
```
Comment (mgdelacroix) [inline commands/command_e2e_test.go:18]: This test case should run for all clients

```suggestion
	s.RunForAllClients("List commands for a non existing team", func(c client.Client) {
```

File: commands/command_e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		defer func() {
+			appErr = s.th.App.DeleteCommand(command2.Id)
+			s.Require().Nil(appErr)
+		}()
+
+		err := listCommandCmdF(s.th.SystemAdminClient, &cobra.Command{}, []string{})
+		s.Require().Nil(err)
+		s.Len(printer.GetLines(), 2)
+		s.ElementsMatch([]*model.Command{command1, command2}, printer.GetLines())
+		s.Len(printer.GetErrorLines(), 0)
```
Comment (mgdelacroix) [inline commands/command_e2e_test.go:124]: This bit should be wrapped in the `s.RunForSysadminAndLocal`

**Review by mgdelacroix — CHANGES_REQUESTED** [review]
Looking really good @SezalAgrawal! Requested a couple of minor changes to include more users in the existing test cases

**Comment by SezalAgrawal** [comment]
Thanks @mgdelacroix for your inputs! I will make the necessary changes.

**Comment by SezalAgrawal** [comment]
@mgdelacroix I have updated the test cases.

**Review by mgdelacroix — APPROVED** [review]
Thanks @SezalAgrawal for applying the changes! LGTM! :sparkles:

**Pull request merged**